### DISCUSSION TITLE
how to normalize the Pole Figure of different crystallographic directions in respect to each other?

### DISCUSSION THREAD

**Discussion by nargesmbg** [mtex-toolbox/mtex#1844] [Ask Anything]
Hello everyone, 

I am making my pole figures in different crystallographic directions (100)(110)and (111). I want to normalize the all three pole figures in respect to each other to have colorbar scaled from 0 to 1 for all of them to be able to compare the results. 
This is how I'm writing my code:

h = [Miller(1,0,0,crystalSymmetry('m-3m', [3.5 3.5 3.5], 'mineral', 'Ni'),'hkl'),Miller(1,1,0,crystalSymmetry('m-3m', [3.5 3.5 3.5], 'mineral', 'Ni'),'hkl'),Miller(1,1,1,crystalSymmetry('m-3m', [3.5 3.5 3.5], 'mineral', 'Ni'),'hkl')]
r = regularS2Grid('resolution',2.5*degree);

odf = calcDensity(ebsd('Ni').orientations);
pf = calcPoleFigure(odf,h,r);
pf_normalized = normalize(pf);

plot(pf_normalized,'contourf');

I tried to use the "normalize()" command but it didn't really work and got the same plot as before. currently the intensity colorbar for: 
1- (100) is scaled from 0 to 3.5 
2- (110) is scaled from 0 to 1.4 
3- (111) is scaled from 0 to 1.8 

I want to have all similarly scaled? 
just for your information I also tried normalize(pf,odf)  command but got the same results. 

Best 
Narges

**Comment by lfgmorales** [comment]
Hi

The density calculation is already normalized, in the case of a uniform distribution, it has a value of 1 everywhere

Now if you want to have all the pole figures with one single colorbar, you can do


mtexColorbar
CLim(gcm,'equal');
mtexColorbar

You can also „force“ the maxima of the contours to be 1, but this would be weird and I believe incorrect, as the meaning of the uniform distribution is to be equal to 1

Cheers

Luiz


Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 30 May 2023, at 16:12, nargesmbg ***@***.***> wrote:

﻿

Hello everyone,

I am making my pole figures in different crystallographic direction (100)(110)and (111). I want to normalize the all three pole figures in respect to each other to have colorbar scaled from 0 to 1 for all of them to be able to compare the results.
This is how I'm writing my code?

h = [Miller(1,0,0,crystalSymmetry('m-3m', [3.5 3.5 3.5], 'mineral', 'Ni'),'hkl'),Miller(1,1,0,crystalSymmetry('m-3m', [3.5 3.5 3.5], 'mineral', 'Ni'),'hkl'),Miller(1,1,1,crystalSymmetry('m-3m', [3.5 3.5 3.5], 'mineral', 'Ni'),'hkl')]
r = regularS2Grid('resolution',2.5*degree);

odf = calcDensity(ebsd('Ni').orientations);
pf = calcPoleFigure(odf,h,r);
pf_normalized = normalize(pf,odf);

plot(pf_normalized,'contourf');

I tried to use the "normalize()" command but it didn't really work and got the same plot as before. currently the intensity colorbar for:
1- (100) is scaled from 0 to 3.5
2- (110) is scaled from 0 to 1.4
3- (111) is scaled from 0 to 1.8

I want to have all similarly scaled?
just for your information I also tried normalize(pf,odf) command but got the same results.

Best
Narges

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1844>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN4NPOCCRJ3OCMWWMETXIX52VANCNFSM6AAAAAAYUCQ3WE>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1844", "url": "https://github.com/mtex-toolbox/mtex/discussions/1844", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]
